### ISSUE TITLE
How to show an additional error message when validation fails on submit

### ISSUE THREAD

**Issue by awakened23** [formium/formik#1294]
## ❓Question
I need to show an additional error message if validation fails on submit. This error message should not be displayed if validation fails on change/ on blur. Looks like there is no way to tell if the validation errors happen on submit since the validate function is called on change, blur and submit. The validation that happens on change/blur can be disabled so that I know the errors happen on submit but I need the validation to happen on change and blur as well.

Here is what I did:
```
<Formik
  validate={values => {
    const errors = {};

    if(!values.reason){
      errors.reason = 'This is required';
    }
    return errors;
  }}
>
  {{errors, status, setStatus, validateForm} => (
    <Form>
      {status && <div>There was a problem submitting the form</div>}
      <Field type="text" name="reason"/>
      <ErrorMessage name="reason" />
      <button type"submit" onClick={() => {
        validateForm().then(errors => {
          if(Object.keys(errors).length > 0){
            setStatus('error');
          }
        });
      }}>
        Submit
      </button>
    </Form>
  )}  
</Formik>
```
This works but the validate function is run twice when the submit button is clicked. Is there a better way to do this?

**Comment by jaredpalmer** [comment]
Not at the moment. This is something to consider for us to consider for v2

**Comment by bratva** [comment]
Yes, i need to different validation on change from validation on submit.
When i have validation from change, i want to validate only touched fields, but before submit - want to validate all fields

**Comment by TaylorDale** [comment]
Good answer at https://github.com/jaredpalmer/formik/issues/1484#issuecomment-490558973

**Comment by Cryfear** [comment]
lol no answer, :(((
